hadoop-hdfs-project/hadoop-hdfs-native-client/src/CMakeLists.txt

```diff
@@ -127,7 +127,8 @@ if(OPENSSL_LIBRARY AND OPENSSL_INCLUDE_DIR)
     include(CheckCSourceCompiles)
     set(OLD_CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES})
     set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
-    check_c_source_compiles("#include \"${OPENSSL_INCLUDE_DIR}/openssl/evp.h\"\nint main(int argc, char **argv) { return !EVP_aes_256_ctr; }" HAS_NEW_ENOUGH_OPENSSL)
```

Comment:
Users can use OpenSSL installed to non-standard location by specifying properties like `-Dopenssl.lib=/path/to/lib -Dopenssl.include=/path/to/include`. 
https://github.com/apache/hadoop/blob/9b1d3579b483069d0a211cb0b29c1f25013684dd/hadoop-hdfs-project/hadoop-hdfs-native-client/pom.xml#L221-L223

Does this change break the feature?

Reply:
Thanks for the explanation. Let me try the patch on Linux env with custom OpenSSL location.